src/qibolab/instruments/icarusqfpga.py

```diff
@@ -346,8 +349,11 @@ def sweep(
 
                 # Since the sweeper will modify the readout pulse serial, we collate the results with the qubit number.
                 # This is only for qibocal compatiability and will be removed with IcarusQ v2.
-                if pulse.type is PulseType.READOUT:
-                    res[pulse.serial] = res[pulse.qubit]
+                # FIXME: if this was required, now it's completely broken, since it
+                # isn't possible to identify the pulse channel from the pulse itself
+                # (nor it should be needed)
+                # if pulse.type is PulseType.READOUT:
+                #     res[pulse.serial] = res[pulse.qubit]
```

Comment:
It should be possible to figure out the type, as explained in the previous comment.

Reply:
I extended the `FIXME` comment. Should I do something else here, or shall I merge?

Reply:
I believe we can merge this.